src/uploaders/NodeUploader.ts

```diff
@@ -0,0 +1,205 @@
+import { promises as fs } from 'fs'
+import path from 'path'
+import http from 'http'
+import readPkgUp from 'read-pkg-up'
+import glob from 'glob'
+
+import { Logger, noopLogger } from '../Logger'
+import request, { PayloadType } from '../Request'
+import AddSources from '../transformers/AddSources'
+import StripProjectRoot from '../transformers/StripProjectRoot'
+import formatErrorLog from './lib/FormatErrorLog'
+import stringifyFileAccessError from './lib/StringifyFileAccessError'
+
+interface UploadSingleOpts {
+  apiKey: string
+  sourceMap: string
+  bundle: string
+  appVersion?: string
+  overwrite?: boolean
+  projectRoot?: string
+  endpoint?: string
+  requestOpts?: http.RequestOptions
+  logger?: Logger
+}
+
+export async function uploadOne ({
+  apiKey,
+  bundle,
+  sourceMap,
+  appVersion,
+  overwrite = false,
+  projectRoot = process.cwd(),
+  endpoint = 'https://upload.bugsnag.com/',
+  requestOpts = {},
+  logger = noopLogger
+}: UploadSingleOpts): Promise<void> {
+  logger.info(`Uploading node source map for "${bundle}"`)
+
+  const [ sourceMapContent, fullSourceMapPath ] = await readSourceMap(sourceMap, projectRoot, logger)
+  const [ bundleContent, fullBundlePath ] = await readBundleContent(bundle, projectRoot, sourceMap, logger)
+
+  const sourceMapJson = parseSourceMap(sourceMapContent, sourceMap, logger)
+  const transformedSourceMap = await applyTransformations(fullSourceMapPath, sourceMapJson, projectRoot, logger)
+
+  if (!appVersion) {
+    appVersion = await detectAppVersion(projectRoot, logger)
+  }
+
+  logger.debug(`Initiating upload "${endpoint}"`)
+  const start = new Date().getTime()
+  try {
+    await request(endpoint, {
+      type: PayloadType.Browser,
```

Comment:
Should we rename this? I didn't realise the payloads for node & browser were the same

Reply:
oh yeah, I missed this. The payloads are indeed the same, but that might not always be true, so worth giving them their own names, even if structurally they are the same type.